src/wazuh_modules/wm_osquery_monitor.c

```diff
@@ -635,10 +660,38 @@ void *wm_osquery_monitor_main(wm_osquery_monitor_t *osquery)
     pthread_join(treader, NULL);
 
     minfo("Closing module.");
+#ifdef WIN32
+    return 0;
+#else
     return NULL;
+#endif
 }
 
+#ifdef WIN32
+DWORD WINAPI wm_osquery_monitor_destroy(void *osquery_monitor)
+{
+    int i;
+    wm_osquery_monitor_t *osquery_monitor_ptr = (wm_osquery_monitor_t *)osquery_monitor;
 
+    if (osquery_monitor_ptr)
+    {
+        free(osquery_monitor_ptr->bin_path);
+        free(osquery_monitor_ptr->log_path);
+        free(osquery_monitor_ptr->config_path);
+
+        for (i = 0; osquery_monitor_ptr->packs[i]; ++i) {
+            free(osquery_monitor_ptr->packs[i]->name);
+            free(osquery_monitor_ptr->packs[i]->path);
+            free(osquery_monitor_ptr->packs[i]);
+        }
+
+        free(osquery_monitor_ptr->packs);
+        free(osquery_monitor_ptr);
+    }
+
+    return 0;
```

Comment:
this could be summarized with a #ifdef WIN32 to avoid the entire duplication of the function.